controller/scheduler-v2/scheduler.go

```diff
@@ -0,0 +1,1199 @@
+		for range ch {
+			s.triggerSyncJobs()
+		}
+	}()
+}
+
+func (s *Scheduler) tickSyncFormations(d time.Duration) {
+	logger.Info("starting sync formations ticker", "duration", d)
+	go func() {
+		ch := time.Tick(d)
+		for range ch {
+			s.triggerSyncFormations()
+		}
+	}()
+}
+
+func (s *Scheduler) triggerRectify() {
+	logger.Info("triggering rectify")
+	select {
+	case s.rectify <- struct{}{}:
+	default:
+	}
+}
+
+func (s *Scheduler) triggerSyncJobs() {
+	logger.Info("triggering job sync")
+	select {
+	case s.syncJobs <- struct{}{}:
+	default:
+	}
+}
+
+func (s *Scheduler) triggerSyncFormations() {
+	logger.Info("triggering formation sync")
+	select {
+	case s.syncFormations <- struct{}{}:
+	default:
+	}
+}
+
+type Stream struct {
+	s      *Scheduler
+	events chan Event
+}
+
+func (s *Stream) Close() error {
+	s.s.Unsubscribe(s.events)
+	return nil
+}
+
+func (s *Stream) Err() error {
+	return nil
+}
+
+func (s *Scheduler) sendEvent(event Event) {
+	s.listenMtx.RLock()
+	defer s.listenMtx.RUnlock()
+	logger.Info(fmt.Sprintf("sending %s event to %d listener(s)", event.Type(), len(s.listeners)), "event", event.Type(), "err", event.Err())
+	for ch := range s.listeners {
+		// drop the event if the listener is too slow to avoid blocking the main loop
+		select {
+		case ch <- event:
+		default:
+		}
+	}
+}
+
+type Event interface {
+	Type() EventType
+	Err() error
+}
+
+type EventType string
+
+const (
+	EventTypeDefault         EventType = "default"
+	EventTypeLeaderChange    EventType = "leader-change"
+	EventTypeClusterSync     EventType = "cluster-sync"
+	EventTypeFormationSync   EventType = "formation-sync"
+	EventTypeFormationChange EventType = "formation-change"
+	EventTypeRectify         EventType = "rectify"
+	EventTypeJobStart        EventType = "start-job"
+	EventTypeJobStop         EventType = "stop-job"
+	EventTypeJobRequest      EventType = "request-job"
+	EventTypeHostEvent       EventType = "host-event"
+)
+
+type DefaultEvent struct {
+	err error
+	typ EventType
+}
+
+func (de *DefaultEvent) Err() error {
+	return de.err
+}
+
+func (de *DefaultEvent) Type() EventType {
+	return de.typ
+}
+
+type JobEvent struct {
+	Event
+	Job *Job
+}
+
+type LeaderChangeEvent struct {
+	Event
+	IsLeader bool
+}
+
+type JobRequestEvent struct {
+	Event
+	Request *JobRequest
+}
+
+func NewEvent(typ EventType, err error, data interface{}) Event {
+	switch typ {
+	case EventTypeJobStop:
+		fallthrough
+	case EventTypeJobStart:
+		job, _ := data.(*Job)
+		return &JobEvent{Event: &DefaultEvent{err: err, typ: typ}, Job: job}
+	case EventTypeJobRequest:
+		req, _ := data.(*JobRequest)
+		return &JobRequestEvent{Event: &DefaultEvent{err: err, typ: typ}, Request: req}
+	case EventTypeLeaderChange:
+		isLeader, _ := data.(bool)
+		return &LeaderChangeEvent{Event: &DefaultEvent{err: err, typ: typ}, IsLeader: isLeader}
+	default:
+		return &DefaultEvent{err: err, typ: typ}
+	}
+}
+
+func getBackoffPeriod() time.Duration {
+	backoffPeriod := time.Second
```

Comment:
`BACKOFF_PERIOD` is not set explicitly to 10m anywhere, we rely on the default value, and only override it in tests.